os/arch/arm/src/armv7-m/up_doirq.c

```diff
@@ -117,13 +117,17 @@ uint32_t *up_doirq(int irq, uint32_t *regs)
 	 * only be modified for outermost interrupt handler (when g_nestlevel == 0)
 	 */
 
-	flags = irqsave();
+	/* To prevent higher priority interrupt corrupt current_regs.
+	 * Moved below code in exception_common under critical area
+	 */
+
+	/* flags = irqsave();
 
 	if (g_nestlevel == 0) {
 		current_regs = regs;
 	}
 
-	g_nestlevel++;
+	g_nestlevel++;*/
```

Comment:
Do we need to keep them with commented?

Reply:
Not sure, I though to keep this code and added the comment stating "To prevent higher priority interrupt corrupt current_regs, Moved below code in exception_common under critical area"

But I am seeking your suggestion on that, what is better to do? remove code or add comment ?

Reply:
I think you can make a separated commit only for this. It is easier to know why we modify this.

Reply:
I mean we could remove or move with separated commit. not for commenting.